pygeos/test/test_io.py

```diff
@@ -248,6 +245,51 @@ def test_to_wkb_srid():
     assert np.frombuffer(result[5:9], "<u4").item() == 4326
 
 
+def test_to_wkb_point_empty():
+    # empty point converts to POINT (nan, nan, nan) in WKB
+    # this matches PostGIS behaviour
+    assert pygeos.to_wkb(empty_point) == POINT_NAN_WKB
+
+
+def test_from_wkb_point_nan():
+    # WKB representation of POINT (nan, nan, nan) converts to POINT EMPTY
+    geom = pygeos.from_wkb(POINT_NAN_WKB)
+    assert pygeos.get_type_id(geom) == 0
+    assert pygeos.is_empty(geom)
+    assert pygeos.get_coordinate_dimension(geom) == 3
+
+
+def test_to_wkb_point_empty_srid():
+    expected = pygeos.set_srid(empty_point, 4236)
+    wkb = pygeos.to_wkb(expected, include_srid=True)
+    actual = pygeos.from_wkb(wkb)
+    assert pygeos.get_srid(actual) == 4236
+    
+
+@pytest.mark.parametrize("expected", [
+    pygeos.multipoints([empty_point]),
+    pygeos.geometrycollections([empty_point]),
+    pygeos.geometrycollections([pygeos.multipoints([empty_point])]),
+    pygeos.geometrycollections([pygeos.geometrycollections([empty_point])])
+])
+def test_to_wkb_roundtrip(expected):
+    wkb = pygeos.to_wkb(expected, include_srid=True)
+    actual = pygeos.from_wkb(wkb)
+    assert pygeos.equals_exact(actual, expected)
+
+
+@pytest.mark.parametrize("geom", [
+    pygeos.multipoints([empty_point]),
+    pygeos.geometrycollections([empty_point]),
+    pygeos.geometrycollections([pygeos.multipoints([empty_point])]),
+    pygeos.geometrycollections([pygeos.geometrycollections([empty_point])])
```

Comment:
this is the same parametrization as the test above, but this one checks that conversion to wkb fails, while the above tests that the roundtrip works ?